core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -55,6 +70,32 @@ public SparkSqlDialect(SqlDialect.Context context) {
       SqlNode fetch) {
     unparseFetchUsingLimit(writer, offset, fetch);
   }
+
+  @Override public void unparseCall(SqlWriter writer, SqlCall call,
+      int leftPrec, int rightPrec) {
+    if (call.getOperator() == SqlStdOperatorTable.SUBSTRING) {
+      SqlUtil.unparseFunctionSyntax(SPARKSQL_SUBSTRING, writer, call);
+    } else {
+      switch (call.getKind()) {
+      case FLOOR:
+        if (call.operandCount() != 2) {
+          super.unparseCall(writer, call, leftPrec, rightPrec);
```

Comment:
Cause your logic has call operands as `!=2` and `=2`, we need to add test cases for them separately.

Reply:
Done. This rewriting logic is similar to that in `PostgresqlSqlDialect` for [CALCITE-1798](https://issues.apache.org/jira/browse/CALCITE-1798). Add a test case for one operand.